drivers/include/periph/hwcrypto.h

```diff
@@ -0,0 +1,401 @@
+ * @return                  HWCRYPTO_ERROR if other error
+ */
+int hwcrypto_cipher_decrypt(hwcrypto_t dev,
+                            const uint8_t *cipher_block,
+                            uint8_t *plain_block,
+                            uint32_t block_size);
+
+/**
+ * @brief   Initialize hardware crypto device for a given hashing algorithm.
+ *
+ * This operation may alter the peripheral's state, therefore it should be used
+ * exclusively.
+ *
+ * @param[in] dev           the device to initialize
+ * @param[in] hash          desired hash algorithm
+ *
+ * @return                  HWCRYPTO_OK on success
+ * @return                  HWCRYPTO_NOTSUP if hash is not supported
+ * @return                  HWCRYPTO_ERROR if other error
+ */
+int hwcrypto_hash_init(hwcrypto_t dev, hwcrypto_hash_t hash);
+
+/**
+ * @brief   Update the digest with a block of data.
+ *
+ * This operation may alter the peripheral's state, therefore it should be used
+ * exclusively.
+ *
+ * A hardware crypto device not initialized for hashing may produce unexpected
+ * results.
+ *
+ * @param[in] dev           the device to initialize
+ * @param[in] block         block of data to update digest with
+ * @param[in] block_size    number of bytes in data block
+ *
+ * @return                  number of bytes added to digest
+ * @return                  HWCRYPTO_NOTSUP if hashing operation not supported
+ * @return                  HWCRYPTO_INVALID if block_size is incorrect
+ * @return                  HWCRYPTO_ERROR if digest could not be updated
+ */
+int hwcrypto_hash_update(hwcrypto_t dev,
+                         const uint8_t* block,
+                         uint32_t block_size);
+
+/**
+ * @brief   Finalize the hash and copy the result to result buffer.
+ *
+ * This operation may alter the peripheral's state, therefore it should be used
+ * exclusively.
+ *
+ * A hardware crypto device not initialized for hashing may produce unexpected
+ * results.
+ *
+ * @param[in] dev           the device to initialize
+ * @param[out] cipher       result output buffer
+ * @param[in] result_size   number of bytes to copy from digest to result
+ *
+ * @return                  number of bytes copied
+ * @return                  HWCRYPTO_NOTSUP if hash is not supported
+ * @returned                HWCRYPTO_INVALID if @p result_size exceeds hash
+ *                          size
+ */
+int hwcrypto_hash_final(hwcrypto_t dev,
+                        uint8_t* result,
+                        uint32_t result_size);
+
+/**
+ * @brief   Get mutually exclusive access to the hardware crypto peripheral.
+ *
+ * In case the peripheral is busy, this function will block until the
+ * peripheral is available again.
+ *
+ * @param[in] dev           the device to initialize
+ *
+ * @return                  HWCRYPTO_OK on success
+ * @return                  HWCRYPTO_ERROR on error
+ */
+int hwcrypto_acquire(hwcrypto_t dev);
```

Comment:
This could be `void hwcrypto_acquire(hwcrypto_t dev)`, same for release.

This would make it consistent with `spi_acquire/release`